src/main/webapp/workspaces/query-selector.js

```diff
@@ -30,9 +31,26 @@ const useOpenClose = props => {
 const QueryCard = props => {
   const { title, onClick, QueryEditor, query } = props
   const [anchorEl, open, handleOpen, handleClose] = useOpenClose(props)
-
+  const [{ active: isDrawing }] = useDrawInterface()
+  const [wasDrawing, setWasDrawing] = useState(false)
+  const drawAnchorEl = useRef(null)
+  useEffect(
+    () => {
+      if (isDrawing) {
+        setWasDrawing(true)
+        handleClose()
+      }
+      if (!isDrawing && wasDrawing) {
+        handleOpen({
+          currentTarget: drawAnchorEl.current,
```

Comment:
Can you use anchorEl here?

Reply:
Unfortunately that element is null when the user clicks Apply or Cancel on the `DrawingMenu` even using a reference directly to the edit button doesn't work.